src/argilla/server/apis/v1/handlers/questions.py

```diff
@@ -19,27 +19,60 @@
 
 from argilla.server.contexts import datasets
 from argilla.server.database import get_async_db
+from argilla.server.models import Question, QuestionType
 from argilla.server.policies import QuestionPolicyV1, authorize
-from argilla.server.schemas.v1.questions import Question
+from argilla.server.schemas.v1.questions import (
+    LabelSelectionSettingsUpdate,
+    QuestionUpdate,
+    TextQuestionSettingsUpdate,
+)
+from argilla.server.schemas.v1.questions import Question as QuestionSchema
 from argilla.server.security import auth
 from argilla.server.security.model import User
 
 router = APIRouter(tags=["questions"])
 
 
-@router.delete("/questions/{question_id}", response_model=Question)
-async def delete_question(
-    *, db: AsyncSession = Depends(get_async_db), question_id: UUID, current_user: User = Security(auth.get_current_user)
-):
+async def _get_question(db: "AsyncSession", question_id: UUID) -> Question:
     question = await datasets.get_question_by_id(db, question_id)
-
-    await authorize(current_user, QuestionPolicyV1.delete(question))
-
     if not question:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail=f"Question with id `{question_id}` not found",
         )
+    return question
+
+
+@router.patch("/questions/{question_id}", response_model=QuestionSchema)
+async def update_question(
+    *,
+    db: AsyncSession = Depends(get_async_db),
+    question_id: UUID,
+    question_update: QuestionUpdate,
+    current_user: User = Security(auth.get_current_user),
+):
+    question = await _get_question(db, question_id)
+
+    await authorize(current_user, QuestionPolicyV1.update(question))
+
+    if question_update.settings is not None:
```

Comment:
Could we use the pydantic discriminator to avoid this logic? I think we're already using for the `QuestionCreate` schema